bot/build.sh

```diff
@@ -196,4 +196,24 @@ echo "                     -- ./create_tarball.sh ${TMP_IN_CONTAINER} ${EESSI_PI
 ./eessi_container.sh "${COMMON_ARGS[@]}" "${TARBALL_STEP_ARGS[@]}" \
                      -- ./create_tarball.sh ${TMP_IN_CONTAINER} ${EESSI_PILOT_VERSION} ${EESSI_SOFTWARE_SUBDIR_OVERRIDE} /eessi_bot_job/${TGZ} 2>&1 | tee -a ${tar_outerr}
 
+function check_tmp_tarball {
+    dir=$1
+    ls ${dir} | grep -E 'tgz$|\.gz$'
+    return $?
+}
+
+# clean storage used on local disk
+if [[ -d ${STORAGE} ]]; then
+    # double-check that at least one tarball of the temporary storage was created
+    if check_tmp_tarball ${TARBALL_TMP_BUILD_STEP_DIR} || check_tmp_tarball ${TARBALL_TMP_TARBALL_STEP_DIR}; then
+        echo "Removing temporary storage under '${STORAGE}'"
+        rm -rf ${STORAGE}
```

Comment:
If I understand things correctly, `$STORAGE` refers to the base directory where all jobs create their unique subdirectory? In that case, shouldn't we only remove `${JOB_STORAGE}` here, as there may be other (running) jobs on the same machine?

Reply:
I think, that's right.

Surprisingly, this `${JOB_STORAGE}` is just created (a subdirectory under `${STORAGE}`) but it is never used.

We should do two things
- replace uses of `${STORAGE}` by `${JOB_STORAGE}` when running `eessi_container.sh` and test properly if that still works (maybe a separate PR would be beneficial)
- replace uses of `${STORAGE}` by `${JOB_STORAGE}` in this PR here such that only job specific job temporary storage is removed